purchase_stock_bl_date/models/stock_picking.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    bl_date = fields.Date(string="B/L Date")
+    force_bl_date = fields.Boolean(string="Force B/L Date")
+
+    def action_toggle_is_locked(self):
```

Comment:
I am not comfortable with this method being the trigger to update the past transactions as the implication is not obvious to the user. How about adding a "Update Valuation" button below B/L Date field, which updates past transactions, and then triggers action_toggle_is_locked() to make itself invisible?

Reply:
I agree with tashiro san suggestion. can you please help to follow up @AungKoKoLin1997 ?